sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -735,7 +750,7 @@ private void deleteById(String containerName, Object id, PartitionKey partitionK
             feedResponse.getContinuationToken(),
             sort);
 
-        return new CosmosPageImpl<>(result, pageRequest, total);
+        return new CosmosSliceImpl<>(result, pageRequest, feedResponse.getContinuationToken() != null);
```

Comment:
One concern I have with this implementation is that sometimes, the continuation token can be null, within the query, despite having more pages. I would like to confirm my thought with @mbhaskar

Reply:
response continuationToken will be null only for the last page.

Reply:
Thanks @mbhaskar !